test/cbmc/include/core_json_contracts.h

```diff
@@ -76,189 +127,126 @@ JSONStatus_t JSON_SearchConst( const char * buf,
+/* *INDENT-ON* */
+;
 
 bool skipEscape( const char * buf,
                  size_t * start,
                  size_t max )
-__CPROVER_requires( 0 < max && max < CBMC_MAX_BUFSIZE )
-__CPROVER_requires( __CPROVER_is_fresh( buf, max ) )
-__CPROVER_requires( __CPROVER_is_fresh( start, sizeof( *start ) ) )
-__CPROVER_assigns( *start )
-__CPROVER_ensures( isBool( __CPROVER_return_value ) )
-__CPROVER_ensures( *start >= __CPROVER_old( *start ) )
-__CPROVER_ensures( ( __CPROVER_old( *start ) < max ) ? ( ( *start <= max ) ) : ( *start == __CPROVER_old( *start ) ) )
-__CPROVER_ensures( __CPROVER_return_value ==> ( __CPROVER_old( *start ) < max ) && ( *start >= __CPROVER_old( *start ) + 1 ) );
+skipContract( buf, max, start )
+/* *INDENT-OFF* */
+ensures( result ==> ( old( *start ) < max ) && ( *start >= old( *start ) + 1 ) )
+/* *INDENT-ON* */
+;
 
 bool skipUTF8( const char * buf,
                size_t * start,
                size_t max )
-__CPROVER_requires( 0 < max && max < CBMC_MAX_BUFSIZE )
-__CPROVER_requires( __CPROVER_is_fresh( buf, max ) )
-__CPROVER_requires( __CPROVER_is_fresh( start, sizeof( *start ) ) )
-__CPROVER_assigns( *start )
-__CPROVER_ensures( isBool( __CPROVER_return_value ) )
-__CPROVER_ensures( *start >= __CPROVER_old( *start ) )
-__CPROVER_ensures( ( __CPROVER_old( *start ) < max ) ? ( ( *start <= max ) ) : ( *start == __CPROVER_old( *start ) ) )
-__CPROVER_ensures( __CPROVER_return_value ==> ( __CPROVER_old( *start ) < max ) && ( *start >= __CPROVER_old( *start ) + 1 ) );
+skipContract( buf, max, start )
+/* *INDENT-OFF* */
+ensures( result ==> ( old( *start ) < max ) && ( *start >= old( *start ) + 1 ) )
+/* *INDENT-ON* */
+;
 
 bool skipDigits( const char * buf,
                  size_t * start,
                  size_t max,
                  int32_t * outValue )
-__CPROVER_requires( 0 < max && max < CBMC_MAX_BUFSIZE )
-__CPROVER_requires( __CPROVER_is_fresh( buf, max ) )
-__CPROVER_requires( __CPROVER_is_fresh( start, sizeof( *start ) ) )
-__CPROVER_requires( outValue == NULL || __CPROVER_is_fresh( outValue, sizeof( *outValue ) ) )
-__CPROVER_assigns( *start )
-__CPROVER_assigns( outValue != NULL: *outValue )
-__CPROVER_ensures( isBool( __CPROVER_return_value ) )
-__CPROVER_ensures( __CPROVER_old( *start ) <= *start )
-__CPROVER_ensures( __CPROVER_old( *start ) < max ? *start <= max : *start == __CPROVER_old( *start ) )
-__CPROVER_ensures( __CPROVER_return_value ==> ( __CPROVER_old( *start ) < max ) && ( *start > __CPROVER_old( *start ) ) && ( ( ( buf[ __CPROVER_old( *start ) ] ) >= '0' ) && ( ( buf[ __CPROVER_old( *start ) ] ) <= '9' ) ) );
-
+skipContract( buf, max, start )
+requires( outValue == NULL || allocated( outValue, sizeof( *outValue ) ) )
+assigns( outValue != NULL: *outValue )
+/* *INDENT-OFF* */
+ensures( result ==> ( old( *start ) < max ) && ( *start > old( *start ) ) && ( ( ( buf[ old( *start ) ] ) >= '0' ) && ( ( buf[ old( *start ) ] ) <= '9' ) ) )
```

Comment:
I get the code reuse with the skipContract macro but I also feel like these contracts are hard to read. Could we achieve similar levels of reuse by using functions instead of macros ?

Reply:
Removed `skipContract`.